src/core/search.rs

```diff
@@ -173,6 +173,10 @@ pub trait FindexSearch<
                 fetch_chains_batch_size,
             )
             .await?;
+
+        // Send current results (Location and NextKeyword) to the callback.
+        let progress_callback_flag = self.progress(&res).await?;
```

Comment:
Not sure about the `progress_callback_flag` name? Could we use something more descriptive? Like `should_continue`?